picard/file.py

```diff
@@ -280,25 +301,25 @@ def _save_and_rename(self, old_filename, metadata):
                 if not tail:
                     head, tail = os.path.split(head)
                 while head and tail:
-                    try:
-                        self._rmdir(head)
-                    except BaseException:
-                        break
+                    self._rmdir(head)
                     head, tail = os.path.split(head)
-            except EnvironmentError:
-                pass
+            except OSError as why:
+                log.warning("Error removing directory: %s", why)
+            except SkipRemoveDir as why:
+                log.debug("Not removing empty directory: %s", why)
         # Save cover art images
         if config.setting["save_images_to_files"]:
             self._save_images(os.path.dirname(new_filename), metadata)
         return new_filename
 
     @staticmethod
     def _rmdir(path):
-        junk_files = (".DS_Store", "desktop.ini", "Desktop.ini", "Thumbs.db")
-        if not set(os.listdir(path)) - set(junk_files):
-            shutil.rmtree(path, False)
+        if os.path.realpath(path) in PROTECTED_DIRECTORIES:
+            raise SkipRemoveDir('%s is a protected directory' % path)
+        elif set(os.listdir(path)) - JUNK_FILES:
+            raise SkipRemoveDir('%s is not empty' % path)
         else:
-            raise OSError
+            shutil.rmtree(path, False)
```

Comment:
According to https://docs.python.org/3/library/shutil.html#shutil.rmtree second argument is False by default, and so `shutil.rmtree(path)` should be enough.

Reply:
Done